src/javaharness/java/arcs/api/PortablePromise.java

```diff
@@ -0,0 +1,26 @@
+package arcs.api;
+
+import java.util.function.Consumer;
+
+/**
+ * Portable representation of the Promise class.
+ */
+public interface PortablePromise<T> {
+ public interface Resolver<T> extends Consumer<T> {
+    default void resolve(T arg) {
+      accept(arg);
+    }
+  }
+
+  public interface Rejector<S> extends Consumer<S> {
+    default void reject(S error) {
+      accept(error);
+    }
+  }
+
+  public interface PortablePromiseExecutor<T> {
+    void doInvoke(Resolver<T> resolve, Rejector<?> reject);
+  }
+
+  void then(Consumer<T> onFulfillment);
```

Comment:
make this chainable, then PortablePromise<T> then(...)

Reply:
(sorry, we talked about it and i forgot)
done now. 
also made `then` implementation simpler and more readable.